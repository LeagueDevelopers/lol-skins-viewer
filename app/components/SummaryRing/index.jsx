import React, { PropTypes } from 'react';
import cx from 'classnames';

import AnimatedNumber from 'components/AnimatedNumber';

import ring from 'static/ring.png';

import style from './index.scss';

const SummaryRing = ({ className, onClick, value, label, children }) =>
  <div className={cx(style.summaryRing, className)} onClick={onClick}>
    <img role="presentation" className={style.bg} src={ring} />
    <AnimatedNumber className={style.value} value={value} />
    <span className={style.text}>{children || label}</span>
  </div>;

SummaryRing.propTypes = {
  className: PropTypes.oneOfType([PropTypes.string, PropTypes.object]),
  value: PropTypes.any,
  label: PropTypes.string,
  onClick: PropTypes.func,
  children: PropTypes.node
};

export default SummaryRing;
