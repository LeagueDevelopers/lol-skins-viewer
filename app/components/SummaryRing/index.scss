.summaryRing {
    flex: 0 0 186px;
    display: flex;
    flex-direction: column;
    text-align: center;
		position: relative;
		font-family: 'LoL Display';
		cursor: default;

		.bg {
			position: absolute;
			width: 100%;
			height: 100%;
		}

    .value {
      margin-top: 25%;
      flex: 0 0 40px;
      background: linear-gradient(to bottom, #cdbe91 0%, #c8aa6e 35%, #765c29 100%);
      -webkit-background-clip: text;
      -webkit-text-fill-color: transparent;
      color: #f0e6d2;
      font-size: 40px;
      font-weight: 700;
      line-height: 42px;
      letter-spacing: 0.05em;
    }

    .text {
      flex: 0 1 auto;
      margin: 20px auto;
      width: 100px;
      color: #a09b8c;
      font-size: 14px;
      font-weight: 700;
      line-height: 16px;
      letter-spacing: 0.075em;
      text-transform: uppercase;
    }
  }
