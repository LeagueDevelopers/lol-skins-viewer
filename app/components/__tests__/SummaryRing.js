import React from 'react';
import { shallow, mount } from 'enzyme';

import SummaryRing from '../SummaryRing';

describe('<SummaryRing />', () => {
  it('should render correctly', () => {
    const labelWrapper = shallow(
      <SummaryRing
        className="test"
        value={9999}
        onClick={() => false}
        label="test1"
      />);
    const childrenWrapper = shallow(
      <SummaryRing
        className="test"
        value={9999}
        onClick={() => false}
      >
        test2
      </SummaryRing>);

    expect(labelWrapper.get(0)).toMatchSnapshot();
    expect(childrenWrapper.get(0)).toMatchSnapshot();
  });

  it('should call onClick when it is clicked', () => {
    const handleClick = jest.fn();

    // Mount because we want to check if click events "bubble" properly
    const wrapper = mount(<SummaryRing onClick={handleClick} />);

    wrapper.simulate('click');
    wrapper.find('.value').simulate('click');
    wrapper.find('.text').simulate('click');

    expect(handleClick).toHaveBeenCalledTimes(3);
  });

  it('should prefer children to label prop', () => {
    const label = 'marksmen';
    const child = 'in2017';

    const wrapper = shallow(<SummaryRing label={label}>{child}</SummaryRing>);

    expect(wrapper.find('.text').text()).toBe(child);
  });
});
