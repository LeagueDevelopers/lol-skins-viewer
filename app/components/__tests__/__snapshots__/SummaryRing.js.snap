exports[`<SummaryRing /> should render correctly 1`] = `
<div
  className="summaryRing test"
  onClick={[Function]}>
  <img
    className="bg"
    role="presentation"
    src="test-file-stub" />
  <AnimatedNumber
    className="value"
    value={9999} />
  <span
    className="text">
    test1
  </span>
</div>
`;

exports[`<SummaryRing /> should render correctly 2`] = `
<div
  className="summaryRing test"
  onClick={[Function]}>
  <img
    className="bg"
    role="presentation"
    src="test-file-stub" />
  <AnimatedNumber
    className="value"
    value={9999} />
  <span
    className="text">
    test2
  </span>
</div>
`;
